testdata/generate.go

```diff
@@ -0,0 +1,245 @@
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	// Generate an RSA 4096 bit pub/private key pair.
+	caPrivateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return "", "", err
+	}
+	// Generate the self-signed cert.
+	certBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivateKey.PublicKey, caPrivateKey)
+	if err != nil {
+		return "", "", err
+	}
+	// Encode certificate in PEM format.
+	caCertPEM := new(bytes.Buffer)
+	if err = pem.Encode(caCertPEM, &pem.Block{Type: "CERTIFICATE", Bytes: certBytes}); err != nil {
+		return "", "", err
+	}
+
+	// Encode private key in PEM format.
+	caPrivateKeyPEM := new(bytes.Buffer)
+	if err := pem.Encode(caPrivateKeyPEM, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(caPrivateKey),
+	}); err != nil {
+		return "", "", err
+	}
+
+	return caCertPEM.String(), caPrivateKeyPEM.String(), nil
+}
+
+// removePemHeaders strips the PEM headers from a certificate so it can be used in an Ownership Voucher.
+func removePemHeaders(pemBlock string) string {
+	pemBlock = strings.TrimPrefix(pemBlock, "-----BEGIN CERTIFICATE-----\n")
+	pemBlock = strings.TrimSuffix(pemBlock, "\n-----END CERTIFICATE-----\n")
+	return pemBlock
+}
+
+// newOwnershipVoucher creates an OV for the device serial which is signed by the vendor's CA.
+func newOwnershipVoucher(serial string, pdcPem []byte, vendorCACert *x509.Certificate, vendorCAPriv *rsa.PrivateKey) (string, error) {
+	currentTime := time.Now()
+	ov := OwnershipVoucher{
+		OV: OwnershipVoucherInner{
+			CreatedOn:        currentTime.String(),
+			ExpiresOn:        currentTime.Add(caExpiry).String(),
+			SerialNumber:     serial,
+			PinnedDomainCert: removePemHeaders(string(pdcPem)),
+		},
+	}
+
+	ovBytes, err := json.Marshal(ov)
+	if err != nil {
+		return "", err
+	}
+
+	signedMessage, err := pkcs7.NewSignedData(ovBytes)
+	if err != nil {
+		return "", err
+	}
+	signedMessage.SetDigestAlgorithm(pkcs7.OIDDigestAlgorithmSHA256)
+	signedMessage.SetEncryptionAlgorithm(pkcs7.OIDEncryptionAlgorithmRSA)
+
+	err = signedMessage.AddSigner(vendorCACert, vendorCAPriv, pkcs7.SignerInfoConfig{})
+	if err != nil {
+		return "", err
+	}
+
+	signedBytes, err := signedMessage.Finish()
+	if err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(signedBytes), nil
+}
+
+// writeFile writes the contents to a new file in the current directory.
+func writeFile(contents []byte, filename string) error {
+	f, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	b, err := f.Write(contents)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Wrote %d bytes to %v\n", b, filename)
+	return nil
+}
+
+func main() {
```

Comment:
nit:
for command line utilities we generally use cobra for all use cases as it allows everything to have the same look and feel 
for cli/ flags / testing

this is pretty trivial so agree it not critical